iocore/utils/I_Machine.h

```diff
@@ -79,11 +81,15 @@ struct Machine {
                     );
   /// @return The global instance of this class.
   static self *instance();
+  bool is_self(const char *name);
+  void insert_id(const char *id);
```

Comment:
I think this implies that we keep "hostnames"  and IPs stores as strings. I think it'd make more sense to have two containers here; One storing hostnames (as strings) and one storing IPs (as IpAddr's). @SolidWallOfCode can probably fill in more, but we have a lot more flexibility if we treat IPs as IpAddr's rather than strings I think.